pkg/bloomgateway/client.go

```diff
@@ -274,12 +307,17 @@ func (c *GatewayClient) serverAddrsForFingerprints(tenantID string, groups []*lo
 		level.Warn(c.logger).Log("msg", "using an inefficient algorithm to determin server addresses for fingerprints", "fingerprints", numFingerprints, "tokens", numTokens)
 	}
 
-	fingerprints := make([]uint64, numFingerprints)
-	addresses := make([][]string, numFingerprints)
-	bufDescs, bufHosts, bufZones := ring.MakeBuffersForGet()
+	fingerprints := fingerprintsPool.Get(numFingerprints).([]uint64)
+	defer fingerprintsPool.Put(fingerprints)
+
+	addresses := addressesPool.Get(numFingerprints).([][]string)
+	defer addressesPool.Put(addresses)
```

Comment:
You cannot both return the slices to the pool and return them in the function call.
The caller would need to be in charge of returning the received buffer to the pool.
Could you remove this in your PR, because I do some refactoring around the grouping of fingerprints by server instance.

Reply:
Done